### PR TITLE
protoc-gen-twirp_typescript: updated to latest (last) version

### PR DESCRIPTION
**Pull request by DGollings**
###### Description of changes
Updated to latest/last version before owner archived repo

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/protoc-gen-twirp_typescript/default.nix
Comment (jojosch) [inline pkgs/development/tools/protoc-gen-twirp_typescript/default.nix:16]: Please keep `subPackages = [ "." ];`, since otherwise the example https://github.com/larrymyers/protoc-gen-twirp_typescript/tree/master/example/cmd/haberdasher will be put into the bin-output:

```
results/protoc-gen-twirp_typescript/
└── bin
    ├── haberdasher
    └── protoc-gen-twirp_typescript
```

**Review by jojosch — CHANGES_REQUESTED** [review]
Thank you for updating `protoc-gen-twirp_typescript` to its latest revision. I've added some comments to your changes. Sorry I've taken to long to review your PR.

-----

Result of `nixpkgs-review pr 198269` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>protoc-gen-twirp_typescript</li>
  </ul>
</details>

File: pkgs/development/tools/protoc-gen-twirp_typescript/default.nix
Comment (DGollings) [reply pkgs/development/tools/protoc-gen-twirp_typescript/default.nix:16]: done

**Comment by DGollings** [comment]
@jojosch resolved your comments, thanks for the review

**Review by jojosch — APPROVED** [review]
LGTM.

Result of `nixpkgs-review pr 198269` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>protoc-gen-twirp_typescript</li>
  </ul>
</details>

File: maintainers/maintainer-list.nix
Comment (Janik-Haag) [inline maintainers/maintainer-list.nix:3380]: please rename this commit to `maintainers: add dgollings`

File: pkgs/development/tools/protoc-gen-twirp_typescript/default.nix
Comment (Janik-Haag) [inline pkgs/development/tools/protoc-gen-twirp_typescript/default.nix:23]:
```suggestion
    platforms = with platforms; [ linux darwin ];
```
this does essentially do the same thing but is easier to read

**Review by Janik-Haag — CHANGES_REQUESTED** [review]
Please rename the update commit to `protoc-gen-twirp_typescript: unstable-2021-03-29 -> unstable-2022-08-14` otherwise this looks good. Thank you for the updated :D 

If this doesn't get updated anymore we should probably mark it as insecure at some point.

**Comment by DGollings** [comment]
@Janik-Haag 

Comments should be resolved

**Comment by Janik-Haag** [comment]
I should have been more thorough, sorry for that. Why are you limiting the platforms in the first place?

**Comment by Janik-Haag** [comment]
Okay so my approach to doing this is just not specifying a `platforms` attribute. In my case usually only `x86_64-linux` if it builds and works fine there I open a PR and if the ci tells me the build breaks on f.e. aarch64-darwin I just exclude it with limiting platforms or marking it as broken. You should mark it as broken if it is a package that's supposed to work on $plaform and limit the platforms to exclude the platform otherwise. 

With go packages you can usually be fairly certain that they build fine on any platform.

**Comment by Janik-Haag** [comment]
Oh and by the way you also have to do a rebase because of the mege conflict in ` maintainers/maintainer-list.nix ` this is normally quite straight forward, sync your github fork, do a git pull in your local repo, then `git rebase master` and then force push the updated version.

**Comment by DGollings** [comment]
should be ok now?

File: maintainers/maintainer-list.nix
Comment (Janik-Haag) [inline maintainers/maintainer-list.nix:4036]: it seems like you accidentally deleted a bracket from the entry above :(

**Review by Janik-Haag — COMMENTED** [review]
Thank you for updating, and sorry for the trouble :see_no_evil: 
I hope my review was still helpful to you, if you want I can fix up the missing bracket. 
Feel free to ping me for reviews in the future and also feel free to ask questions :D

File: maintainers/maintainer-list.nix
Comment (DGollings) [reply maintainers/maintainer-list.nix]: gah :D, now?

**Comment by Janik-Haag** [comment]
> > Thank you for updating, and sorry for the trouble see_no_evil I hope my review was still helpful to you, if you want I can fix up the missing bracket. Feel free to ping me for reviews in the future and also feel free to ask questions :D
> 
> It was, thanks, if I were to complain about anything it would be a) this PR was submitted like 7 months ago and worse b) for software that will likely never be updated again, so it doesn't feel all that useful :)

a) I started doing nix like ~8 or 9 months ago so :shrug:  and sadly we don't have enough people reviewing PRs :/ 
as I said feel free to ping me in the future, I try to work through the pile of emails I get in a timely manner.

b) Doesn't really matter to me as long as you learned something.

**Comment by Janik-Haag** [comment]
There is still a small thing left todo which is: 
image
squashing the two commits that are not the maintainer one or renaming the one that removes the platform limit either is fine with, after that (and once ci ran trough) I will merge this.

**Comment by Janik-Haag** [comment]
Do you need some help with squashing the commits?

**Comment by DGollings** [comment]
Oh sorry, I must have either not seen this or forgot to push at the time. On holiday right now, I'll try to remember to do so once I get back

**Comment by Janik-Haag** [comment]
Hope you had some nice holidays :D 
Can we get those commits squashed :point_right: :point_left:

**Comment by DGollings** [comment]
They were excellent, hope the same for you. I pushed just now, hope I read your request correctly :)

**Review by Janik-Haag — COMMENTED** [review]
Nice to hear that :D 

You are almost done, https://github.com/NixOS/nixpkgs/pull/198269/commits/9218d1700a7b2fbe1b50e711495b2a3bd8d0806f also includes the actual package update please make it only include the changes to `maintainers/maintainer-list.nix`

The easiest way to fix this is probably to do a `git reset --mixed 9fadaa2576d10be07f48c9a5eaa02579f2339354` and then `git add maintainers/maintainer-list.nix` and git commit -m "maintainers: add dgollings" and then add the other changes and also do a new commit for them.

**Comment by DGollings** [comment]
weird, you can rebase in any order, but squashes still happen in the same order. I get the logic but its still confusing. Ok, did a reset and commit manually
@Janik-Haag

**Comment by Janik-Haag** [comment]
Result of `nixpkgs-review pr 198269` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>protoc-gen-twirp_typescript</li>
  </ul>
</details>

**Comment by Janik-Haag** [comment]
I didn't want to bother you any further so I fixed the capitilization of the maintainers commit and did a miner formatting change in the package. Should be good to go and I'll merge once ci passes. Hope that's fine with you?

**Comment by Janik-Haag** [comment]
It's a nitpick anyways so :shrug: and basically every pr has a lowercase commit message for the maintainer and it's lowercase in the maintainers-list.nix

**Comment by DGollings** [comment]
agreed, and its all fine, just know it was intentional and not simply trying to increase your workload for no reason

**Comment by Janik-Haag** [comment]
> agreed, and its all fine, just know it was intentional and not simply trying to increase your workload for no reason

No worries :D 
Thank you for your contribution.

**Pull request merged**
